### ISSUE TITLE
Digits/caffe cannot find cuda device in ubuntu 16.04 + cuda 8.0

### ISSUE THREAD

**Issue by mhmorta** [NVIDIA/DIGITS#1169]
Hi,
I have Ubuntu 16.04 and installed cuda 8. I think they are installed correctly because:
![screenshot from 2016-10-16 11-20-38](https://cloud.githubusercontent.com/assets/6433593/19416053/b330b274-9392-11e6-8dbf-3b93341d215e.png)
![screenshot from 2016-10-16 11-21-10](https://cloud.githubusercontent.com/assets/6433593/19416052/b32efdee-9392-11e6-8b28-b784bf33eeb8.png)
but I have problems like this:
![screenshot from 2016-10-16 11-22-10](https://cloud.githubusercontent.com/assets/6433593/19416058/03be221c-9393-11e6-8909-f0b2bb063b8e.png)

And i tried digits/test-devices.py
![screenshot from 2016-10-16 11-23-05](https://cloud.githubusercontent.com/assets/6433593/19416057/039c08b2-9393-11e6-8c74-223577bbf7bc.png)

I should mention that at first it had problems with pydot, so according to previous issues I installed pydot 1.0.28 and pydotparser ver. 1.5.7 ... 

But the problem is I don't know how to resolve problem with cuda.
and these are my environmental variables:
![screenshot from 2016-10-16 11-26-25](https://cloud.githubusercontent.com/assets/6433593/19416068/89d2fd78-9393-11e6-9660-a128825bfb05.png)

So I'd be grateful if you help me ...

**Comment by lukeyeager** [comment]
Does `nvidia-smi` work?

**Comment by mhmorta** [comment]
Yes, and this is the result:
![screenshot from 2016-10-17 21-29-59](https://cloud.githubusercontent.com/assets/6433593/19449048/ebd28aac-94b0-11e6-9f89-ad64d8addfe4.png)

**Comment by lukeyeager** [comment]
Here's what error 38 means:

> cudaErrorNoDevice = 38
This indicates that no CUDA-capable devices were detected by the installed CUDA driver.
http://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__TYPES.html#group__CUDART__TYPES_1g3f51e3575c2178246db0a94a430e0038

... but that's not super helpful. Your GPU has compute capability 3.0, so you should be fine. [This SO post](http://stackoverflow.com/a/17236325/2404152) suggests restarting. Did you install your driver recently?

**Comment by lukeyeager** [comment]
Ping? Any updates?

**Comment by mhmorta** [comment]
Sorry... 
Yes I tried multiple times of restarting and different versions of nvidia driver which the newest was 370, but I couldn't fix that.
It happened that I tried Xubuntu 16.04 on my laptop, but same problem... and same tries and no success!

**Comment by lukeyeager** [comment]
My guess is that it's some sort of hardware issue because you're on a laptop with a relatively old GPU. But I don't know of any specific reason why you should be having an issue. Let me ask around and see if anyone has any suggestions...

**Comment by mhmorta** [comment]
Thanks for your response.
There is something else I should mention, now I'm using cuda V. 8.0.44 + nvidia 370 on Xubuntu and I see the error 35 which is related to the driver.
I should mention I have restarted the laptop.

Thank you so much :)

**Comment by lukeyeager** [comment]
@3XX0's best guess is that this is an Optimus issue. Apparently Canonical installs some problematic packages on Ubuntu that jack with Optimus-enabled GPUs.

Sorry, but this is not a DIGITS issue and is outside the realm of our expertise. I'm going to have to refer you to https://devtalk.nvidia.com. Somebody there should be able to help you out.

**Comment by mhmorta** [comment]
Hi,
   I want to send an update; Now I'm available to train on the same device , I installed everything related to Caffe, Digits, Nvidia driver and Cuda from scratch; Unfortunately, I don't know what was the reason for last failures but I should mention that previously I used cuda-8.0.27 and I realized it's not compatible with new kernel of Ubuntu 16.04 and now 8.0.44 is working with Nvidia-367 driver.
   I can say that I followed some discussions about error code 35 or 38 (rises in Caffe's log), and they didn't work for me. Maybe if something same occurs for you, after double checking, start from the scratch because of the potential missing.
